### PR TITLE
Feature/nex 150 update code from old repo

### PR DESCRIPTION
**Pull request by btamas**
Jira: https://oat-sa.atlassian.net/browse/NEX-150

Update code from `extension-tao-tests`

Test: `npm run test`
Test new ES6 coverage measurement: `npm run test:cov` then `npm run coverage:html`

Related PR:
- [ ] https://github.com/oat-sa/extension-tao-test/pull/290

Task after merge:
- [x] Release it to npm

### CODE REVIEW

File: src/runnerComponent.js [hunk 1]
```diff
@@ -17,148 +17,135 @@
...
-    _.forEach(providers, function(provider) {
-        runnerFactory.registerProvider(provider.name, provider);
-    });
-    return providers;
+function getSelectedProvider(type = 'runner', config = {}) {
+    if (config.provider && config.provider[type]) {
+        return config.provider[type];
+    }
+
+    if (config.providers && config.providers[type]) {
```
Comment (ferenckiss89) [inline src/runnerComponent.js:62]: if (!config.providers || !config.providers[type]) {
    return false;
}
const typeProviders = config.providers[type];
if (typeof typeProviders === 'object' && (typeProviders.id || typeProviders.name)) {
    return typeProviders.id || typeProviders.name;
}
if (Array.isArray(typeProviders) && typeProviders.length > 0) {
    return typeProviders[0].id || typeProviders[0].name;
}

File: test/mocks/mockAltRunnerProvider.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+define([], function () {
```
Comment (ferenckiss89) [inline test/mocks/mockAltRunnerProvider.js:19]: license header is missing

File: test/mocks/mockInvalidRunnerProvider.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+define([], function () {
```
Comment (ferenckiss89) [inline test/mocks/mockInvalidRunnerProvider.js:19]: missing license header

File: test/mocks/mockPlugin2.js [hunk 4]
```diff
@@ -1,16 +1,13 @@
-define([], function () {
```
Comment (ferenckiss89) [inline test/mocks/mockPlugin2.js]: missing license header

File: test/mocks/mockProxyProvider.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+define([], function () {
```
Comment (ferenckiss89) [inline test/mocks/mockProxyProvider.js:19]: missing license header

File: test/mocks/mockRunnerProvider.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+define([], function () {
```
Comment (ferenckiss89) [inline test/mocks/mockRunnerProvider.js:19]: missing license header

**Review by ferenckiss89 — COMMENTED** [review]
First round is done on my side, only code style check and test running. Little comments.

**Review by ferenckiss89 — APPROVED** [review]
Ok for me.

**Review by ferenckiss89 — APPROVED** [review]

File: test/providerLoader/test.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8" />
+        <title>Test Runner - Runner Component</title>
```
Comment (jsconan) [inline test/providerLoader/test.html]: Wrong title, should be `Test Runner - Provider Loader`

**Review by jsconan — COMMENTED** [review]
Looks good.
Only have one remark, nothing blocking.

I also noticed some new unit tests are converted to ES6, while the majority is ES5. That should not be a big issue, but would prevent to run those tests on older browsers (only source code is transpiled, not tests). Anyway, this is not something to change, just a wondering.

File: test/providerLoader/test.html (hunk 7 above)
Comment (btamas) [reply test/providerLoader/test.html]: I just copied the file, but I totally agree. Fixed.

**Comment by btamas** [comment]
> Looks good.
> Only have one remark, nothing blocking.
> 
> I also noticed some new unit tests are converted to ES6, while the majority is ES5. That should not be a big issue, but would prevent to run those tests on older browsers (only source code is transpiled, not tests). Anyway, this is not something to change, just a wondering.

Based on @krampstudio 's presentation last week, we can use ES6 in tests with knowing it won't work in old browsers. It is an acceptable side effect.
The reason why I modified only some part of the tests, because I fixed `eslint` errors what was easy to fix and I did not want to spend time to refactor everything, because it was not part of this task.

**Review by jsconan — APPROVED** [review]
A little typo in the changed title, but anyway :)

**Pull request merged**